### ISSUE TITLE
[Feature Request] Allow to disable "Position icon while moving" altogether

### ISSUE THREAD

**Issue by kssi** [osmandapp/OsmAnd#15260]
### Version

OsmAnd 4.2.4 on iOS 15.6.1

### Problem

OsmAnd has (and has always had for as long as I can remember) a very hard time figuring out whether I'm standing still or not. This means that when I need the "Position icon at rest" the most, e.g. to know which direction my device is pointing toward, I will most likely get a barely usable "Position icon while moving" jiggling on the map. This happens in all conditions, even in open fields with strong GPS signal.

### Solution

I think this issue has been reported 4 years ago here: #6207, but it has yet to be solved.

In the meantime, it would be really nice to have a per-profile setting that allows one to disable heading arrows (aka. "Position icon while moving") altogether. Especially for those like me, who have never found a good use for this feature.

Please make it optional!

**Comment by sonora** [comment]
Issue 6207 had to do with 0/0 bearing reports and to my knowledge is fixed, though we never closed it.

In order to find the root caaue for the issue you report here, could you place your device on a desk and record a track (at rest) so we can check what's going on? I am wondering what your typical hdop values are,  and if your device reports a >0 speed all the time or not.

**Comment by kssi** [comment]
> In order to find the root caaue for the issue you report here, could you place your device on a desk and record a track (at rest) so we can check what's going on? I am wondering what your typical hdop values are, and if your device reports a >0 speed all the time or not.

On a desk as in "indoors"? (In which case I'll most likely get a very poor GPS signal, if any).

But yes, my intuition is that the algorithm that triggers the icon is too sensitive and gets fooled by the device constantly trying to recalculate its position.

Putting this aside, I still think "Position icon while moving" should be optional as it has always been a nuisance for me, especially for low speed activities such as pedestrian hiking.

**Comment by sonora** [comment]
Unless you g to a windowless basement, most hardware can give reasonable reception near a window. I any case, even with borderline reception, it would be interesting to see if your chipset produces bogus positive speed values all the time - it shouldn't. Maybe you really have just a hardware issue?

**Comment by kssi** [comment]
@sonora, please rest assured that if it was an hardware issue, I would have addressed my report to Apple, not OsmAnd devs.

Having clarified that, there are two different topics being discussed here.

### First

Consumer grade GPS never give the exact same position twice as they have a precision of a few meters. As a result, OsmAnd fails to tell appart what is _actual movement_ from what is _usual GPS noise_. This is just how it works, and it more often than not gives OsmAnd the false impression I'm moving whereas I'm not (see the short video below). I have already shared my thoughts above, but I think this should be further discussed in a separate issue if there is any willing to do so, as this is not entirely related to my feature request.

https://user-images.githubusercontent.com/26198013/189546721-450977cb-d00c-4fc5-8f9d-d99e8aa2814a.mov

### Second

The feature request: for my use case, the "Position icon while moving" is an unnecessary feature that causes more trouble than it solves. I never need an arrow icon to assist me in figuring out which direction I'm moving to (especially when it's unreliable as shown above). But I always need to know **which direction my device is pointing to**, regardless of whether I'm moving or not. Just like Google Maps does. Currently, OsmAnd forces me to use the "Position icon while moving" and I have no control over that. I'd rather always have the regular icon with the blue halo that makes use of the much more reliable electronic compass of my device. Since we all have different uses of OsmAnd, I think being able to disable the "Position icon while moving" would be really nice.

**Comment by sonora** [comment]
Plesae do provide a gpx recording, though, because the arrow icon is not assigned based on location fluctuation but based on  the chipset reporting a non zero speed, and that in my experience is quite an anomaly at rest.

**Comment by vshcherb** [comment]
Moving or not moving is determined by GPS + Android API

**Comment by kssi** [comment]
> Regarding "First": Please do provide a gpx recording, though, because the arrow icon is not assigned based on location fluctuation but based on the chipset reporting a non zero speed, and that in my experience is quite an anomaly at rest.

Here is the GPX recording you requested, along with two screenshots illustrating the issue:
[2022-09-19_12-40_lun..zip](https://github.com/osmandapp/OsmAnd/files/9601246/2022-09-19_12-40_lun.zip)

> Regarding "_Second_": I will leave that point to product management if we need yet another parameter and setting for this purpose,or how much we are willing to adopt google maps here ... ;)

I'm a strong supporter of open source myself, and I should probably have expected this reaction from bringing the G-word into the conversation, but please let's not sacrifice useful and _harmless_ features on the altar of ideology... ;)

**Comment by sonora** [comment]
Thanks for the track! Here is a brief analysis:
- It contains 24 recorded points, covering 2:12 minutes, one every 5.7 seconds, no gap.
- It reports hdop accuracies in the interval **[4.9 meters, 13.9 meters], average 5.18 meters, most frequently reported value is 4.7 meters**. Not excellent for outdoors, but reasonable.
- There are indeed 5 points (21% !) which claim non-zero speed value, That's quite a notable weakness assuming the track was recorded at rest.

All Android devices I have (I run them with _Location source=Android API_) behave significantly more benign. For direct comparison I just ran a test indoors(!) on a mid-range (200$) Samsung device and have the following findings:
- 66 points recorded at rest and indoor (GPS only, no other location source), covering 6:01, one point every 5.55 sec.
- hdop accuracies are from the interval **[2 meters, 7.5 meters], average 3,67 meters, most frequently reported value is 2.5 meters**.
- There is only one point reported with a (bogus) non-zero speed recorded
- I watched during the recording and _even after a location fluctuation, OsmAnd never showed an arrow icon_. Instead, the 'at rest' position icon simply 'jumped' a little. (I probably missed the one occasion where actually a speed value was recorded).
- So it's as I said: The arrow icon by our Android code denotes a speed was detected, and does not appear for location fluctuations at rest. Adequate behavior, I would say.

For iOS, I would assume it's the same (but we can check the code): I think the arrow is shown when your device reports these bogus speed values, in your case over 20% of the time! I still wonder if the quality of your GPS reception is the norm for that device or a hardware/firmware issue, have zero experience with iPhones and cannot say.

@vshcherb Regarding code changes, I see no necessity to change anything for Android, iOS remains to be seen?

**Comment by craigerl** [comment]
I did quite a bit of searching trying to disable the "position icon when moving", The video above says it all.  When walking it blinks on and off at random, in random directions.  Making the blue direction cone, which is actually accurate, hard or impossible to see.   Exacerbated by low gps signal, but generally bad.

**Comment by limex** [comment]
Any progress on this?
I understand the the direction arrow makes sense for profiles like car, motorcycle, ... where a clear direction is given based on the speed of movement. Especially when the device might be showing not in the direction of movement.

But: For profiles like hiking, where someone can point in in any direction by free hand. it is not needed.
Or definitely annoying while standing or going slow, as the video perfectly shows.

I guess osmand is the only Mapping App (that I know) with this strange behaviour.

Suggestion: 
Add an "Off" to the "Position icon while moving" setting. 
Or only show the arrow when there is a "clear" or "better" indication of the movement.

**Comment by vshcherb** [comment]
CC @sonora 
We can probably disable that icon if speed < 1 kmh but on most of the devices it's automatically done by GPS sensor itself. It looks strange and not reproducible on my S21

**Comment by kssi** [comment]
> Add an "Off" to the "Position icon while moving" setting.

Yes please.

**Comment by limex** [comment]
@sonora 

> (I have looked through some of my tracks and very occasionally do find speed values like 0.2, or 0.22, but would not consider them artifacts, nor issues for my work with OsmAnd.)

thanks for your analysis. Can I ask you what kind of tracks you consider your "work"? Slow velocity like hiking?

**Comment by FloLan1** [comment]
I would also appreciate being able to turn the moving symbols off.

**Comment by limex** [comment]
> I would also appreciate being able to turn the moving symbols off.

Agree. As long as this is not fixed, osmand is useless for hiking. Therefore I canceled the Pro Subscription.

**Comment by sonora** [comment]
> thanks for your analysis. Can I ask you what kind of tracks you consider your "work"? Slow velocity like hiking?

I mark geologic features (their direction and expanse), usually by walking along them and recording. Could somehow on my devices never detect the bug you are reporting here.

**Comment by mnalis** [comment]
> Could somehow on my devices never detect the bug you are reporting here.

It seems clear to me that the problem is happening more to some people than others. It could be specific phone hardware, it could be android version, it could be manufacturers extensions (GNSS corrections, filtering etc) it could be situation on the ground (e.g. forest canopy etc), it could be how users are moving etc. Or most likely, combination of those. Even if one (or few) of the root causes could be precisely determined, would that really change the situation?

I too think that having per-profile preference is the best way to go, as there does not seem to be _one-size-fits-all solution_ (if the "user is not moving" threshold is set high enough to accomodate hikers with the issue, it will be too high for users that prefer directional display at low speeds). 

What might be decided is whether:
- that preference should be just another answer to _"Position icon while moving"_ (like _"Off"_ or _"Same as position icon at rest"_), or
- extra-techy decimal number preference like _"Detect movement threshold (meters/second)"_ should be introduced (which users could set to "0.5" or "2" to fine tune the detection for their device and use case, or _"999"_ or some other unrealistically high value to practically accomplish the same as _"Off"_ option above)

I'd personally prefer the former, as it is simpler (the other would need a FAQ entry to explain that behaviour).

**Comment by syneoa** [comment]
Sometimes it makes you wonder about developers of this app ever making any hikes in their life as this bug is so annoying and it just blews my mind it's not fixed yet.
Please remove this atrocious icon so I can enjoy my hikes without triangle obstacle blocking my direction view.

**Comment by limex** [comment]
@syneoa 
I agree that this is an annoying bug for hikers, therefor also for me.
But: You don't pay for the software, so you should not blame "developers" for having other priorities.

**Comment by syneoa** [comment]
I did pay Maps+ package of OsmAnd thinking that I would have all features forever but alas, it keeps asking me to go to subscription model which I hate but I digress

Anyway this is not the point and I am not demanding anything, I am just asking please add "None" to the box bellow, Thanks!

![image](https://github.com/osmandapp/OsmAnd/assets/72040612/63bd4633-e082-4993-a14a-551ff6c93a12)

**Comment by vshcherb** [comment]
No icon means nothing will be displayed doesn't make much sense to me

**Comment by kssi** [comment]
@limex

> But: You don't pay for the software, so you should not blame "developers" for having other priorities.

Making the software _usable_ should be a top priority, being paid or not. It's just common sense.

@vshcherb 

I would happily stick to the electronic compass (circle with halo). I ended up stopping using OsmAnd because of this bug and I'm hoping it will become an option soon. Smart workaround on top of a handy feature.

**Comment by limex** [comment]
> No icon means nothing will be displayed doesn't make much sense to me

True. When I stand still I want "Show beacon with the **view** direction"

**Comment by syneoa** [comment]
> No icon means nothing will be displayed doesn't make much sense to me

No icon means no position triangle icon shown but you still have blue dot showing you at which direction  your phone "looks". Just what we need when we hike.

**Comment by vshcherb** [comment]
To do: this pull request should be migrated to iOS https://github.com/osmandapp/OsmAnd/pull/19393
